engine/src/main/java/org/terasology/engine/rendering/assets/mesh/ObjMeshFormat.java

```diff
@@ -50,63 +48,50 @@ public MeshData load(ResourceUrn urn, List<AssetDataFile> inputs) throws IOExcep
                 throw new IOException("No index data");
             }
 
-            StandardMeshData data = processData(rawVertices, rawNormals, rawTexCoords, rawIndices);
-
-            if (data.getVertices() == null) {
-                throw new IOException("No vertices define");
-            }
-            if (!data.normals.isEmpty() && data.normals.size() != data.getVertices().size()) {
-                throw new IOException("The number of normals does not match the number of vertices.");
-            }
-            if (!data.uv0.isEmpty() && data.uv0.size() / 2 != data.getVertices().size() / 3) {
-                throw new IOException("The number of tex coords does not match the number of vertices.");
-            }
-
-            return data;
+            return processData(rawVertices, rawNormals, rawTexCoords, rawIndices);
         }
     }
 
     private StandardMeshData processData(List<Vector3f> rawVertices, List<Vector3f> rawNormals, List<Vector2f> rawTexCoords, List<Vector3i[]> rawIndices) throws IOException {
+        int numIndices = 0;
+        int numVerts = 0;
+        for (int x = 0; x < rawIndices.size(); x++) {
+            numIndices += (rawIndices.get(x).length - 2) * 3;
+            numVerts += rawIndices.get(x).length;
+        }
```

Comment:
How often is this code called/how performance critical is this? I personally would prefer using a more functional approach using the Streams API:

```java
        final int numVerts = rawIndices.stream().map(x -> x.length).reduce(0, Integer::sum);
        final int numIndices = rawIndices.stream().map(x -> (x.length - 2) * 3).reduce(0, Integer::sum);
```

And instead of doing the computation for `numIndices` inside the loop we could do the math and only do it at the end, can't we? This would only do one stream iteration then.

```java
final int numIndices = (numVerts - 2 * rawIndices.size()) * 3;
```

Reply:
umm not that much

Reply:
lets move this to another PR. its non critical to how this is suppose to work.